phpunit/class-block-library-comment-template-test.php

```diff
@@ -282,4 +282,90 @@ function test_build_comment_query_vars_from_block_sets_cpage_var() {
 		$this->assertEquals( $actual['paged'], $comment_query_max_num_pages );
 		$this->assertEquals( get_query_var( 'cpage' ), $comment_query_max_num_pages );
 	}
+
+	/**
+	 * Test that unapproved comments are included if it is a preview.
+	 */
+	function test_build_comment_query_vars_from_block_with_comment_preview() {
+		$parsed_blocks = parse_blocks(
+			'<!-- wp:comment-template --><!-- wp:comment-author-name /--><!-- wp:comment-content /--><!-- /wp:comment-template -->'
+		);
+
+		$block = new WP_Block(
+			$parsed_blocks[0],
+			array(
+				'postId' => self::$custom_post->ID,
+			)
+		);
+
+		$commenter_filter = function () {
+			return array(
+				'comment_author_email' => 'unapproved@example.org',
+			);
+		};
+
+		add_filter( 'wp_get_current_commenter', $commenter_filter );
+
+		$this->assertEquals(
+			build_comment_query_vars_from_block( $block ),
+			array(
+				'orderby'       => 'comment_date_gmt',
+				'order'         => 'ASC',
+				'status'        => 'approve',
+				'no_found_rows' => false,
+				'include_unapproved' => array( 'unapproved@example.org' ),
+				'post_id'       => self::$custom_post->ID,
+				'hierarchical'  => 'threaded',
+				'number'        => 5,
+				'paged'         => 1,
+			)
+		);
+
+		remove_filter( 'wp_get_current_commenter', $commenter_filter );
```

Comment:
The test suite removes filters as part of the default tear down so `remove_filter()` can be removed from this test.

Tell me to shut the heck up if Gutenberg differs to WP Core.

Reply:
Haha, I'm not sure. 😄  I just searched in the [wordpress-develop](https://github.com/WordPress/wordpress-develop/search?q=remove_filter&type=code) repository, and it looks like a lot of PHPUnit tests are adding and also removing the filters they need.

If you mean the `WP_UnitTestCase::tear_down()` function, correct me if I'm wrong, but isn't that function being called right after executing all tests? If that's the case, we would need to remove the filter anyway because it could affect the following tests in the test suite. Maybe it is safer to leave the `remove_filter` after all; I don't know. 🤔

Reply:
Yes, they're reset in `WP_UnitTestCase_Base::tear_down()` which calls `WP_UnitTestCase_Base::_restore_hooks()`

`tear_down` is called between _each_ test so there's not chance of polluting other tests. You're right, `remove_filter()` in a number of tests within core but the calls are periodically removed https://github.com/WordPress/wordpress-develop/commit/2ef42cb78c8544ae38089452ceb9c58b402b7d68

Reply:
Okay, thank you very much @peterwilsoncc, I'll remove it then. 😊